### PR TITLE
rewrite to single python script

### PR DESCRIPTION
**Pull request by yuanji-dev**
According to the discussion here  https://github.com/Ajatt-Tools/transformers_ocr/pull/3#issuecomment-1538368189, I tried to do a PoC, seems it's possible to merge the logic from the bash file into a single Python file. I did some rough testing, and it seems to be working fine.

### CODE REVIEW

**Comment by tatsumoto-ren** [comment]
Thanks. I'll try it out. 

Maybe it would be better to split the file into smaller modules? I think having everything in one file harms readability.

File: src/transformers_ocr.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+            ],
+            check=True,
+        )
+
+
+def prepare_pipe():
+    if os.path.exists(PIPE_PATH):
+        os.unlink(PIPE_PATH)
+    if not os.path.exists(PIPE_PATH):
+        os.mkfifo(PIPE_PATH)
```
Comment (tatsumoto-ren) [inline src/transformers_ocr.py]: This function will delete the pipe every time it is run.
The original function looked like this:
```
def prepare_pipe():
    if os.path.isfile(PIPE_PATH):
        os.remove(PIPE_PATH)
    if not is_fifo(PIPE_PATH):
        os.mkfifo(PIPE_PATH)
```

Comment (tatsumoto-ren) [reply src/transformers_ocr.py]: I have replaced the function with the original version because otherwise it would cause freezes.

**Comment by yuanji-dev** [comment]
> Maybe it would be better to split the file into smaller modules? I think having everything in one file harms readability.

Personally, I feel the size of the current script is fine to put all in one file, of course, if needed you can split it into smaller ones. The main reason I rewrote it into Python is to avoid the same logic written in both Bash & Python(eg: notify_send, prepare_pipe, some consts and the potential config file parse logic).

One single file maybe has another benefit, if users want to try this project, they just need to download a file and run it. About the readability maybe we can add some comment sections in the file just like https://github.com/bottlepy/bottle/blob/master/bottle.py

**Comment by tatsumoto-ren** [comment]
After some testing, I think it works as intended now. There's still some refactoring better be done to improve readability though. Do you notice any problems? If not, I'll merge.

**Comment by yuanji-dev** [comment]
LGTM, thank you!

**Comment by yuanji-dev** [comment]
BTW, @tatsumoto-ren 
I'm curious about which tool did you use to record your screen displayed in README.md, I noticed there is a zoom-in effect, was it done during the screencast phase, or was added by another tool afterwards?

**Comment by tatsumoto-ren** [comment]
>  which tool did you use to record your screen displayed in README.md

I used [dmenurecord](https://github.com/tatsumoto-ren/dotfiles/blob/main/.local/bin/dmenurecord).

> there is a zoom-in effect, was it done during the screencast phase, or was added by another tool afterwards?

I edited the video afterwards in kdenlive.

**Pull request merged**